sdk/log/provider.go

```diff
@@ -123,26 +134,72 @@ func NewLoggerProvider(opts ...LoggerProviderOption) *LoggerProvider {
 
 // Logger returns a new [log.Logger] with the provided name and configuration.
 //
+// If p is shut down, a [noop.Logger] instace is returned.
+//
 // This method can be called concurrently.
 func (p *LoggerProvider) Logger(name string, opts ...log.LoggerOption) log.Logger {
-	// TODO (#5060): Implement.
-	return &logger{}
+	if name == "" {
+		global.Warn("Invalid Logger name.", "name", name)
+	}
+
+	if p.stopped.Load() {
+		return noop.NewLoggerProvider().Logger(name, opts...)
+	}
+
+	cfg := log.NewLoggerConfig(opts...)
+	scope := instrumentation.Scope{
+		Name:      name,
+		Version:   cfg.InstrumentationVersion(),
+		SchemaURL: cfg.SchemaURL(),
+	}
+
+	p.loggersMu.Lock()
+	defer p.loggersMu.Unlock()
+
+	if p.loggers == nil {
```

Comment:
Can we simply move the loggers initialization to NewLoggerProvider?

Reply:
That will allocate a map with there are no processors.

Reply:
> If I am not wrong, this would not allocate the map only if `Logger` method is never called for a `LoggerProvider` which before it was shutdown. This is a corner case.

Right, that was what I was trying to say before (:facepalm:)

Reply:
The main issue with only having the map allocated in `NewLoggerProvider` is that the `LoggerProvider` is exported. A user can create one with `var _ LoggerProvider{}`.

I think it is fair to have zero-value behavior there (i.e. invalid limits and no processors), but I'm hesitant to have the type panic if `Logger` is called.

I can abstract this initialization to an `init` function so it can be reused in the future if that makes more sense to you.

Reply:
TIL. I  checked Go's standard library source code and I see that almost all exported types are not panicking for zero value. https://pkg.go.dev/time#Timer is the most popular exception from that rule but I have not found any other popular struct that has invalid zero value. Thank you for being assertive.

Resolving the comment.